### PR TITLE
Add support for v1alpha2

### PR DESCRIPTION
**Pull request by mcbenjemaa**
Towards #29 

This PR adds:
- Add support for v1alpha2
- Support DHCP in the v1alpha2 APIs
- Conversation webhook

### CODE REVIEW

**Review by avorima — COMMENTED** [review]
I like that we finally made all import aliases for the API package similar.

Seeing how little the CRDs actually changed, I wonder whether a new API version is actually needed. If you would've made all DHCP changes in v1alpha1 there would be no difference to v1alpha2, from a YAML perspective. They all have `omitempty` and are disabled by default which is the same data and behavior as now.
Sure, the code changes are breaking but since this is alpha I don't think they should have the same guarantees as the user-facing YAML.

What do you think, do you want to continue with v1alpha2 or do we accept a breaking code change in favor of simplicity?

File: internal/webhook/proxmoxmachine_webhook.go
[the remark this answers is not in the document]
Comment (mcbenjemaa) [reply internal/webhook/proxmoxmachine_webhook.go:1]: yes

File: pkg/scope/cluster.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
-	infrav1alpha1 "github.com/ionos-cloud/cluster-api-provider-proxmox/api/v1alpha1"
+	infrav1alpha1 "github.com/ionos-cloud/cluster-api-provider-proxmox/api/v1alpha2"
```
[the remark this answers is not in the document]
Comment (mcbenjemaa) [reply pkg/scope/cluster.go:31]: infrav1

**Comment by mcbenjemaa** [comment]
> I like that we finally made all import aliases for the API package similar.
> 
> Seeing how little the CRDs actually changed, I wonder whether a new API version is actually needed. If you would've made all DHCP changes in v1alpha1 there would be no difference to v1alpha2, from a YAML perspective. They all have `omitempty` and are disabled by default which is the same data and behavior as now. Sure, the code changes are breaking but since this is alpha I don't think they should have the same guarantees as the user-facing YAML.
> 
> What do you think, do you want to continue with v1alpha2 or do we accept a breaking code change in favor of simplicity?

As stated in the issue #29 

This will change the type of object to IPConfig, which is a breaking change, but you're right we can support this even in v1alpha1 as long as the yaml is not a breaking change.

```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: ProxmoxCluster
metadata:
  name: "test-cluster"
spec:
  controlPlaneEndpoint:
    host: 10.10.10.3
    port: 6443
  ipv4Config:
    dhcp: true
  dnsServers: [10.1.1.1]
---
## OR

apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: ProxmoxCluster
metadata:
  name: "test-cluster"
spec:
  controlPlaneEndpoint:
    host: 10.10.10.3
    port: 6443
  ipv4Config:
      dhcp: false
      addresses: ["10.10.10.5-10.10.10.60"]
      prefix: 24
      gateway: 10.10.10.1
  dnsServers: [10.1.1.1]
```

**Comment by avorima** [comment]
The examples mentioned in the issue include this `static` field that would indeed be breaking.

```yaml
  ipv4Config:
    static:
      addresses: ["10.10.10.5-10.10.10.60"]
      prefix: 24
      gateway: 10.10.10.1
```

The dependency on the IPAM struct was a mistake in hindsight, so replacing it with a custom struct would not be something requires us to do a new API version. If we were in this position from the start we would't even have considered a v1alpha2.

I would favor keeping v1alpha1.

**Comment by mcbenjemaa** [comment]
This PR will now be closed, until the plans for v1alpha2

**Comment by mcbenjemaa** [comment]
This PR will now be closed, until the plans for v1alpha2

**Pull request closed without merging**